client/dashboard/sites/site/environment-switcher.tsx

```diff
@@ -58,55 +65,69 @@ const CurrentEnvironment = ( { site }: { site: Site } ) => {
 	return <Environment env="production" />;
 };
 
+const StagingSiteActionButton = ( {
+	isStagingSiteDeleting,
+	isStagingSiteCreating,
+}: {
+	isStagingSiteDeleting: boolean;
+	isStagingSiteCreating: boolean;
+} ) => {
+	if ( isStagingSiteCreating ) {
+		return (
+			<>
+				<Spinner style={ { width: '24px', height: '24px', padding: '4px', margin: 0 } } />
```

Comment:
Nit: Since the same style is applied to both Spinner kinds, maybe we could extract it out to a constant.